docs/api.rst

```diff
@@ -600,6 +608,8 @@ section generates a new *URL* instance.
       URL('http://example.com/path?a=b&c=d')
       >>> URL('http://example.com/path?a=b').update_query({'c': 'd'})
       URL('http://example.com/path?a=b&c=d')
+      >>> URL('http://example.com/path?a=b').update_query({'c': [1, 2]})
+      URL('http://example.com/path?a=b&c=1&c=2')
```

Comment:
This may be seen as controversial because, for example, web-browsers use `varname[]` format for arrays:
```diff
      URL('http://example.com/path?a=b&c=1&c=2')
      URL('http://example.com/path?a=b&c[]=1&c[]=2')
```

Reply:
Does that not depend on the REST API? Users can use `'c[]'` if the API expects arrays to use `'c[]'` as the key, although I haven't seen that.

Once again, `'c'` and not `'c[]'` is consistent with both `urllib` and `requests`, and sites like Google translate don't use `dt[]` but `dt`.

Reply:
I think this depends on how the backend software handles this. PHP used to handle it gracefully (the form sends `c[]` but then in HTTP handler it's already `c` with values).

You should at least add a good tests coverage including such corner cases.

Reply:
Could you suggest a way to test this? We already have tests that cover the fact it's `c` and not `c[]`. I will try to write some test page and server to see what browsers send.

Reply:
Just add a test with `{'c[]': [1, 2]}` making sure that `[]` is preserved non-encoded.